subsys/net/ip/tcp.c

```diff
@@ -417,7 +417,9 @@ int net_tcp_prepare_segment(struct net_tcp *tcp, u8_t flags,
 
 	if (flags & NET_TCP_FIN) {
 		tcp->flags |= NET_TCP_FINAL_SENT;
-		seq++;
+printk("Making sure FIN goes with ACK; adjusting seq by 2; now fire up Wireshark and check that all's ok\n");
+		flags |= NET_TCP_ACK;
+		seq += 2;
```

Comment:
That "magic extra byte" in sequence numbers for SYN and FIN packets is really annoying.  But I agree with Jukka: it'd be better to implement this as an "increment because we received a FIN" and not a "+2" for clarity.

Also, a warning: I ran into a spot (I forget the precise details) where wireshark was lying to me about packet parsing.  It was making assumptions based on the ethernet frame that happened to be wrong, or something.  Use tcpdump to verify before making a change based solely on wireshark. :)